src/trip_kinematics/URDFParser.py

```diff
@@ -0,0 +1,301 @@
+    from_urdf() to build a tree of joints.
+
+    Args:
+        joints (List[ET.Element]): List of <joint> tags from the URDF file.
+
+    Returns:
+        dict[str, Dict]: A dictionary representing parent-child relationships between joints.
+    """
+    # Keep track of parent and children links of joints
+    # and use that to search for the parent joint of each joint
+    joint_tree_dict = {
+        joint.get('name'): {
+            'child_link': joint.find('child').get('link'),
+            'parent_link': joint.find('parent').get('link')
+        } for joint in joints}
+
+    parent_link_to_joint = defaultdict(list)
+    child_link_to_joint = defaultdict(list)
+
+    for joint in joints:
+        parent_link_to_joint[joint.find('parent').get('link')].append(joint.get('name'))
+        child_link_to_joint[joint.find('child').get('link')].append(joint.get('name'))
+
+    for val in joint_tree_dict.values():
+        val['parent'] = child_link_to_joint[val['parent_link']]
+        val['child'] = parent_link_to_joint[val['child_link']]
+
+    return joint_tree_dict
+
+
+def _get_transformations_for_joint(joint: ET.Element) -> List[List]:
+    """Generates the parameters for the transformations for the input joint. One joint is
+    represented by up to five transformations. These are:
+
+        1. origin translation and 2. origin rotation (both taken from the <origin> tag of the URDF),
+        3. a rotation that makes joint movement parallel to the z axis of the local coord. system,
+        4. the joint movement (the only non-fixed transformation of the five),
+        5. the inverse transformation of number 3.
+
+    Args:
+        joint (ET.Element): A <joint> tag in the URDF file.
+
+    Raises:
+        ValueError: Could not parse URDF file.
+
+    Returns:
+        List[List]: A list of parameters for up to four py:class`Transformation` objects that
+        describe the input joint.
+    """
+    # Read properties from urdf
+    name = joint.get('name')
+    type_ = joint.get('type')
+    origin = joint.find('origin')
+    joint_transformations = []
+
+    # File needs to specify a joint type for each joint
+    try:
+        assert type_ is not None
+    except AssertionError as err:
+        raise ValueError(f'Error: Invalid URDF file ({err})') from err
+
+    # Default values if origin rotation or translation are not specified
+    if origin is None:
+        xyz_vals = '0 0 0'
+        rpy_vals = '0 0 0'
+    else:
+        xyz_vals = origin.get('xyz')
+        rpy_vals = origin.get('rpy')
+        if xyz_vals is None:
+            xyz_vals = '0 0 0'
+        if rpy_vals is None:
+            rpy_vals = '0 0 0'
+    origin_xyz = np.array(list(map(float, xyz_vals.split(' '))))
+    origin_rpy = np.array(list(map(float, rpy_vals.split(' '))))
+
+    if type_ in ['fixed']:
+        axis = None     # Fixed joints have no axis
+
+    elif type_ in ['continuous', 'revolute', 'prismatic']:

```

Comment:
using an elif implies a final else case.
floating and  planar joints should raise an error

Reply:
Edit: you seem to raise this value later on line 241. Try to raise all type errors at the top of the function this decouples type checks from the flow logic and makes it easier to see which errors are checked.